### PR TITLE
First home fix Vitalik Ischenko

### PR DESCRIPTION
**Pull request by futurestark**

### CODE REVIEW

**Comment by futurestark** [comment]
sorry for been late

File: firstHomeAdvanced.cpp [hunk 1]
```diff
@@ -0,0 +1,289 @@
...
+
+	//-------------------------------------- variant by merge algorithm  --------------------------------
+	
+	std::merge(firstVecBeg, firstVecEnd,
+		secondVecBeg, secondVecEnd,
+		std::back_inserter(combinedVector));   // a esli vmesto std::back_inserter(combinedVector));  peredat prosto combinedVector.begin()  to runtime fail  why??####
+												// v prototype merge 4 alocatora i posledniy out dest parametr, no prosto combinedVector on but ne mozhet ############
+
+	//--------------------------------- variant assign one vector in another is not concationation !! ---
+				//	combinedVector.assign(10, 1); // çàïîëíåíèå âåêòîðà èç 10 ýëåìåíòîâ ÷èñëîì 1
```
Comment (smay1613) [inline firstHomeAdvanced.cpp:113]: use English even in comments

File: firstHomeAdvanced.cpp [hunk 2]
```diff
@@ -0,0 +1,289 @@
...
+
+	//-------------------------------------- variant by copying -----------------------------------------
+	//std::copy(firstVecBeg, firstVecEnd, std::back_inserter(combinedVector));
+	//std::copy(secondVecBeg, secondVecEnd, std::back_inserter(combinedVector));
+
+	//-------------------------------------- variant by merge algorithm  --------------------------------
+	
+	std::merge(firstVecBeg, firstVecEnd,
+		secondVecBeg, secondVecEnd,
+		std::back_inserter(combinedVector));   // a esli vmesto std::back_inserter(combinedVector));  peredat prosto combinedVector.begin()  to runtime fail  why??####
```
Comment (smay1613) [inline firstHomeAdvanced.cpp:109]: because it's capacity is 0

File: firstHomeAdvanced.cpp [hunk 3]
```diff
@@ -0,0 +1,289 @@
...
+
+			flagCycleBegin = true;
+
+		}
+
+		return valuesList;
+	}
+}
+
+std::forward_list<int> reverseFrowardList(std::forward_list<int> & sourceForwardList) {      //peredalat 4erez ssilku
```
Comment (smay1613) [inline firstHomeAdvanced.cpp:174]: **this method should have 'void' return type and modify parameter**

File: firstHomeAdvanced.cpp [hunk 4]
```diff
@@ -0,0 +1,289 @@
...
+	}
+	//for (auto it = newforwardlist.begin(); it != newforwardlist.end(); ++it)
+	//	std::cout << ' ' << *it;
+
+	return  newforwardlist;
+}
+
+
+/*
+	Ïî ïîâîäó ïåðâîãî äç, ÿ îæèäàþ îò âàñ òðè ôóíêöèè:
```
Comment (smay1613) [inline firstHomeAdvanced.cpp:190]: why this comments are here??
what the mess.

**Pull request closed without merging**